src/traffic_server/traffic_server.cc

```diff
@@ -901,6 +916,141 @@ cmd_verify(char * /* cmd ATS_UNUSED */)
   return 0;
 }
 
+enum class plugin_type_t {
+  GLOBAL,
+  REMAP,
+};
+
+/**
+ * @brief Create an appropriate PluginInfo object.
+ *
+ * @param[in] args The arguments passed to the -C command option. This includes
+ * verify_global_plugin.
+ *
+ * @param[in] plugin_type The type of plugin for which to create a PluginInfo.
+ * @param[in] plugin_path That path to the plugin's shared object file.
+ * @param[in] runtime_directory The directory the PluginInfo will use to
+ *   verify the plugin.
+ *
+ * @return a PluginInfo object.
+ */
+static std::function<bool(std::string &)>
```

Comment:
Why are you returning the function to load the plugin instead of loading it in this function and returning a bool?

Reply:
Yeah, I like what you're suggesting. I'll switch it to that.

In an earlier version of this review, I wrote this as a factory that would return a base PluginInfo object (either RemapPluginInfo or a GlobalPluginInfo, the latter of which I wrote in a previous version of this review). Then the caller would dispatch a common `load` virtual function which would polymorphically call the appropriate load logic. Gancho, however, has future plans for global plugin loading and desired for me to hold off on starting the GlobalPluginInfo implementation at the moment, which is fine. For now I switched the function to return a load function rather than a non-existent PluginInfo base class, thinking we could just easily switch things out for the base PluginInfo object in the future. But I like what you're suggesting (calling the appropriate logic underneath). That will fit both scenarios.

Reply:
> Gancho, however, has future plans for global plugin loading and desired for me to hold off on starting the GlobalPluginInfo implementation at the moment, which is fine

For the record I liked @bneradt's polumorphically calling the corresponding load logic creating `GlobalPluginInfo` and subclassing `PluginDso`. 

My concern was more about the scope of this PR. In this case we were about to change **inadvertently** the current global and remap plugin loading behavior just to take advantage of C++ polymorphic capabilities (by side effect combining the global and the remap plugins DSO loading which is currently completely isolated).
 
My initial suggestion was just to reuse the existing core validation APIs and we already have a C API for global plugins and C++ API for remap plugins.

IMO "merging" global and remap plugins needs much more careful consideration than to do it by "side-effect" as part of this PR which goal is to expose the plugin validation to the command-line.